sound/soc/sof/ipc4-topology.c

```diff
@@ -2009,28 +2009,32 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 		copier_data = &ipc4_copier->data;
 		available_fmt = &ipc4_copier->available_fmt;
 
-		/*
-		 * Use the fe_params as a base for the copier configuration.
-		 * The ref_params might get updated to reflect what format is
-		 * supported by the copier on the DAI side.
-		 *
-		 * In case of capture the ref_params returned will be used to
-		 * find the input configuration of the copier.
-		 */
-		ref_params = kmemdup(fe_params, sizeof(*ref_params), GFP_KERNEL);
-		if (!ref_params)
-			return -ENOMEM;
-
-		ret = sof_ipc4_prepare_dai_copier(sdev, dai, ref_params, dir);
-		if (ret < 0)
-			return ret;
+		if (dir == SNDRV_PCM_STREAM_PLAYBACK) {
+			/*
+			 * For playback the pipeline_params needs to be used to
+			 * find the input configuration of the copier.
+			 */
+			ref_params = kmemdup(pipeline_params, sizeof(*ref_params),
+					     GFP_KERNEL);
```

Comment:
How does this take into consideration any SRC, up/down mixer in the pipeline post copier ? i.e. does this always assume no format/rate change in the pipeline ?

Reply:
@lgirdwood, the pipeline_params is the param at the output of the previous module, it is refined throughout the path from host and if there were SRC or other modules that modified it then it is reflected at this point.